src/main/mainWindow.ts

```diff
@@ -499,5 +502,13 @@ export async function createWindows() {
         });
     });
 
+    mainWin.webContents.on("did-navigate", (_, url: string, responseCode: number) => {
```

Comment:
why is this necessary?

Reply:
Seems like it might somehow prevent an infinitely looping/refreshing webpage. But it's unclear why it is needed yeah.

The event triggers when the page URL changes (not just anchors but actually navigation to another page):

https://www.electronjs.org/docs/latest/api/web-contents#event-did-navigate

And if this code detects that the new page is not starting with `/app` and the HTTP status code was 300 or higher (errors), it sets the URL to `undefined` to kill the browser instead of letting it loop on an error page.

That's as far as I got but not really sure *why* this is needed.